src/producer.rs

```diff
@@ -26,7 +26,8 @@ use util::cstr_to_owned;
 
 /// A native rdkafka-sys topic, used for message production.
 struct NativeTopic {
-    ptr: *mut RDKafkaTopic,
+    client_ptr: *mut RDKafka,
```

Comment:
This line now contains a pointer to an `RDKafka`, which wasn't there before. However it's not a big deal, i'm working on a branch that reorganizes the way this part of the code works. The new version should be more straightforward.